### PR TITLE
ERC20 dynamic decimals 

### LINKED ISSUE #150: Allow more than 4 dp for ERC20 token
```
 I notice the decimal limit is 4, but it will need to be whatever the token they’re entering supports (18 decimal places for ETH for instance)
```

### LINKED ISSUE #185: Pressing "MAX" button with balance `<0.0001` crashes app
Error: `Error: [big.js] Invalid number`

https://user-images.githubusercontent.com/52016903/166522464-8e1bf3c1-3265-47e6-8618-53574eee1188.mov

### PR DESCRIPTION
**Pull request by eupston**
## Description

close #150
close #185 

Updates erc20 token value displayed to dynamically account for token specific decimals

### Changes

- updates swap/pool/bridge value displayed to account for token decimal number

Screen Shot 2022-05-03 at 4 33 19 PM
Screen Shot 2022-05-03 at 4 33 29 PM

### CODE REVIEW

**Review by aidan-starke — APPROVED** [review]
lgtm!

**Comment by aidan-starke** [comment]
I noticed #185 while testing it out - pretty sure the issue was there already but maybe you could cover it in this PR?

**Comment by ken-centrality** [comment]
yeah unfortunately it's a not as simple as changing the `scale` property, #185 was the reason why I didn't do it in the beginning. Have some ideas on how to fix it, will take a look today and give you more details @eupston

**Comment by ken-centrality** [comment]
@eupston another issue is when you type in the input field, say `.0000005` it will switch to `57`, because it's switched to scientific notation `5e-7`

Screen Shot 2022-05-04 at 14 47 41


https://user-images.githubusercontent.com/97480229/166617105-ace137f5-2119-47c1-8402-88cbea67c5ef.mov

**Comment by ken-centrality** [comment]
okay @eupston @aidan-starke made a few more changes to this PR, fix the issue I reported early. Pretty happy with it, can you guys please give it a spin, and do actual transactions too please.

Screen Shot 2022-05-04 at 17 39 37

**Comment by eupston** [comment]
yup looks good on all components! 
Also nice too in the pool, cause if you enter a very small ETH amount it converts to under 0.0 CPAY and the form won't allow you to submit, which will prevent failed transaction.

File: components/BridgeProgress.tsx [hunk 1]
```diff
@@ -136,7 +136,7 @@ const TxFailure: VFC<IntrinsicElements["div"] & TxFailureProps> = ({
 			<h1>Transaction Failed</h1>
 			<p>
 				An error occurred while processing your transaction. It might have gone
-				through, check your balances before trying again.
+				through, check your balances before try again.
 			</p>
```
Comment (aidan-starke) [inline components/BridgeProgress.tsx:140]: this was fine as is, or you could say `check your balances before you try again`

File: components/PoolProgress.tsx [hunk 2]
```diff
@@ -124,7 +124,7 @@ const TxFailure: VFC<IntrinsicElements["div"] & TxFailureProps> = ({
 			<h1>Transaction Failed</h1>
 			<p>
 				An error occurred while processing your transaction. It might have gone
-				through, check your balances before trying again.
+				through, check your balances before try again.
 			</p>
```
Comment (aidan-starke) [inline components/PoolProgress.tsx:128]: same, revert or `check your balances before you try again`

File: components/SwapProgress.tsx [hunk 3]
```diff
@@ -121,7 +121,7 @@ const TxFailure: VFC<IntrinsicElements["div"] & TxFailureProps> = ({
 			<h1>Transaction Failed</h1>
 			<p>
 				An error occurred while processing your transaction. It might have gone
-				through, check your balances before trying again.
+				through, check your balances before try again.
 			</p>
```
Comment (aidan-starke) [inline components/SwapProgress.tsx:125]: same typo

**Review by aidan-starke — CHANGES_REQUESTED** [review]
Looking good, but sadly pressing MAX on with balance `<0.001` on `swap` still crashes the app.  

works on `pool` though!

Screen Shot 2022-05-05 at 6 39 12 AM

**Comment by aidan-starke** [comment]
nice @eupston pressing MAX works, but maybe too well now lol

I was able to submit this transaction, and it said it succeeded, but it [failed](https://uncoverexplorer.com/extrinsic/0x57863f79e9f32625070eab3e7ab505f2b580c7bff56c48bfddb045db30acd239?network=Nikau)

Screen Shot 2022-05-05 at 11 44 46 AM

**Comment by eupston** [comment]
@aidan-starke yes thats the last thing we need to sort is prevent TX for 18 decimal tokens to 4 decimal tokens when the value of opposing token is less then 4 decimal places (0.0001). 

On the pool this works great cause when the value goes under 0.0001 the **min** or **empty** field kicks in preventing the form from being submitted. But on the Swap these don't seem to be kicking in yet.

**Comment by ken-centrality** [comment]
hmm I still want to have that `>0.0001` display in certain cases. Anyway, tell you what let's merge this in as it's stable enough @eupston and I'll create another PR to bring `<0.0001` back without crashing the `swap`

**Review by ken-centrality — APPROVED** [review]

**Review by aidan-starke — APPROVED** [review]

**Pull request merged**